kotlin-power-assert-plugin/build.gradle.kts

```
@@ -30,6 +44,25 @@ tasks.withType<KotlinCompile> {
 
 tasks.withType<Test> {
   useJUnitPlatform()
+  systemProperty("idea.ignore.disabled.plugins", "true")
+  systemProperty("idea.home.path", project.rootDir)
+  doFirst {
+    setLibraryProperty("org.jetbrains.kotlin.test.kotlin-stdlib", "kotlin-stdlib")
+    setLibraryProperty("org.jetbrains.kotlin.test.kotlin-stdlib-jdk8", "kotlin-stdlib-jdk8")
+    setLibraryProperty("org.jetbrains.kotlin.test.kotlin-reflect", "kotlin-reflect")
+    setLibraryProperty("org.jetbrains.kotlin.test.kotlin-test", "kotlin-test")
+    setLibraryProperty("org.jetbrains.kotlin.test.kotlin-script-runtime", "kotlin-script-runtime")
+    setLibraryProperty("org.jetbrains.kotlin.test.kotlin-annotations-jvm", "kotlin-annotations-jvm")
+
+    setLibraryProperty("org.opentest4j.opentest4j", "opentest4j")
+    setLibraryProperty("org.junit.jupiter.junit-jupiter-api", "junit-jupiter-api")
+    setLibraryProperty("org.junit.platform.junit-platform-commons", "junit-platform-commons")
+  }
+}
+
+tasks.create<JavaExec>("generateTests") {
+  classpath = sourceSets.test.get().runtimeClasspath
+  mainClass.set("com.bnorm.power.runners.GenerateTestsKt")
```

This could enable caching, up-to-date, and provide separation:
```suggestion
val generateTests = tasks.register<JavaExec>("generateTests") {
  classpath = sourceSets.test.get().runtimeClasspath
  mainClass.set("com.bnorm.power.runners.GenerateTestsKt")
  inputs.dir(layout.dir("src/test/data"))
        .withPropertyName("testData")
        .withPathSensitivity(PathSensitivity.RELATIVE)
  outputs.dir(layout.dir("src/test/gen"))
         .withPropertyName("generatedTests")
}
sourceSets.test.configure { java.srcDir(generateTests) }
```
(Untested, and might need a bit different syntax for Kotlin (`kotlin.complations.etc`).)